File: 3719_Longest_Balanced_Subarray_I/solution.py
class Solution:
    def longestBalanced(self, nums: List[int]) -> int:
        n, result = len(nums), 0
        for i in range(n):
            even = set()
            odd = set()
            for j in range(i, n):
                if nums[j] % 2 == 0:
                    even.add(nums[j])
                else:
                    odd.add(nums[j])
                if len(even) == len(odd):
                    result = max(result, j - i + 1)
        return result

File: 3719_Longest_Balanced_Subarray_I/solution.c
int longestBalanced(int* nums, int numsSize) {
    int result = 0;
    for (int i = 0; i < numsSize; i++) {
        int evenCount = 0, oddCount = 0, even[numsSize], odd[numsSize];
        for (int j = i; j < numsSize; j++) {
            bool found = false;
            if (nums[j] % 2 == 0) {
                for (int k = 0; k < evenCount; k++) if (even[k] == nums[j]) { found = true; break; }
                if (!found) even[evenCount++] = nums[j];
            } else {
                for (int k = 0; k < oddCount; k++)  if (odd[k] == nums[j]) { found = true; break; }
                if (!found) odd[oddCount++] = nums[j];
            }
            if (evenCount == oddCount) { int length = j - i + 1; if (length > result) result = length; }
        }
    }
    return result;
}
